pkg/serverless/invocationlifecycle/lifecycle.go

```diff
@@ -262,3 +339,45 @@ func (lp *LifecycleProcessor) setParentIDForMultipleInferredSpans() {
 	lp.requestHandler.inferredSpans[1].Span.ParentID = lp.requestHandler.inferredSpans[0].Span.ParentID
 	lp.requestHandler.inferredSpans[0].Span.ParentID = lp.requestHandler.inferredSpans[1].Span.SpanID
 }
+
+type spanTagSetter RequestHandler
+
+func (s *spanTagSetter) SetMeta(tag string, value string) {
+	s.triggerTags[tag] = value
+}
+
+func (s *spanTagSetter) SetMetrics(tag string, value float64) {
+	s.triggerMetrics[tag] = value
+}
+
+func makeAppSecContext(path *string, headers, queryParams map[string][]string, pathParams map[string]string, sourceIP string, body string) *httpsec.Context {
```

Comment:
This could live in the `appsec` package

Reply:
Done in https://github.com/DataDog/datadog-agent/pull/14185/commits/f646ba9a61739331e4f1767f9cb47e8f4a8cd3f5